src/nat_cmd.c

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <modem/at_cmd.h>
+#include <modem/lte_lc.h>
+#include <net/socket.h>
+#include <shell/shell.h>
+#include <shell/shell_uart.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <zephyr.h>
+
+#include "nat_test.h"
+
+#define WAIT_TIME_S 3
+#define THREAD_STACK_SIZE 8192
+#define AT_LOG_TIMEOUT_S 20
+
+static void handle_at_cmd(const struct shell *shell, size_t argc, char **argv)
+{
+	int err;
+	char response[200];
+	enum at_cmd_state state;
+
+	if (argc <= 1) {
+		shell_print(shell, "AT command was not provided\n");
+		return;
+	}
+
+	err = at_cmd_write(argv[1], response, sizeof(response), &state);
+	if (err < 0) {
+		shell_print(shell, "Error while processing AT command: %d\n",
+			    err);
+		state = AT_CMD_ERROR;
+	}
+
+	switch (state) {
+	case AT_CMD_OK:
+		shell_print(shell, "%s\n", response);
+		shell_print(shell, "OK\n");
+		break;
+	case AT_CMD_ERROR:
+		shell_print(shell, "ERROR\n");
+		break;
+	default:
+		break;
+	}
+}
+
+SHELL_CMD_REGISTER(at, NULL, "AT commands", handle_at_cmd);
+
+static void handle_set_timeout(const struct shell *shell, size_t argc,
+			       char **argv)
+{
+	long value;
+	char *unused;
+
+	if (argc <= 1) {
+		shell_print(shell, "Timeout value was not provided\n");
+		return;
+	}
+
+	value = strtol(argv[1], &unused, 10);
+	if (value <= 0) {
+		shell_print(shell, "Timeout value needs to be > 0\n");
+	}
+
+	if (!strcmp(argv[-2], "udp")) {
+		udp_initial_timeout = value;
+		shell_print(shell, "UDP timeout multiplier set to: %d",
+			    udp_initial_timeout);
+	} else if (!strcmp(argv[-2], "tcp")) {
+		tcp_initial_timeout = value;
+		shell_print(shell, "TCP timeout multiplier set to: %d",
+			    tcp_initial_timeout);
+	}
+}
+
+static void handle_get_timeout(const struct shell *shell, size_t argc,
+			       char **argv)
+{
+	if (!strcmp(argv[-2], "udp")) {
+		shell_print(shell, "UDP initial timeout: %d\n",
+			    udp_initial_timeout);
+	} else if (!strcmp(argv[-2], "tcp")) {
+		shell_print(shell, "TCP initial timeout: %d\n",
+			    tcp_initial_timeout);
+	}
+}
+
+static void handle_set_multiplier(const struct shell *shell, size_t argc,
+				  char **argv)
+{
+	float value;
+	char *unused;
+
+	if (argc <= 1) {
+		shell_print(shell, "Multiplier value was not provided\n");
+		return;
+	}
+
+	value = strtof(argv[1], &unused);
```

Comment:
I think you can pass `NULL` as end pointer

Reply:
True. Changed it.